common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -122,6 +125,9 @@ public class WebViewAuthorizationFragment extends AuthorizationFragment {
     // This is used by the switch browser protocol to handle the resume of the flow.
     private SwitchBrowserProtocolCoordinator mSwitchBrowserProtocolCoordinator = null;
 
+    private boolean isBrokerRequest = false;
+    private boolean isEstsRequest = false;
```

Comment:
TBH, I don't think we make requests anywhere besides eSTS so all of our requests are to eSTS. So not sure what this check is doing

Reply:
May be you intend to check for the "current" URL that we are navigating to within the webview? I think that would be ideal but I don't think your code is doing that at the moment

Reply:
No i don't think it's doing that either. I'm not sure if this check is really necessary honestly. The whole point is to make sure the JS API will not be called by any malicious actors, but my understanding is that we already have validation that makes it so our webview is only launched in ESTS pages, and unless they redirect onto a 3rd party page where a malicious actor may have access, then our JS api should be secure. Let me know if this is correct.

Reply:
Idea was that if we validate webview is originally being launched on an ests page, then we should be safe from there, assuming ests is not actively redirecting us to malicious pages. I think this section will need some further consideration.

Reply:
I don't think eSTS will redirect to malicious page, however, we should still limit the usage of this API from just the eSTS page itself (if possible). So if eSTS redirects to ADFS or any third party page then this JS API shouldn't be available to that 3p domain.

Reply:
Moving the javascript exposure to onPageLoaded()